### DISCUSSION TITLE
How to upgrade/patch guest macOS VM to latest version on Apple Silicon M1 / M2 Mac

### DISCUSSION THREAD

**Discussion by hanselsin** [utmapp/UTM#4328] [UTM Help (macOS)]
Apple released macOS 12.5.1 a couple of days ago, of course I patched all of my Macs including dev VMs. But then as macOS VM on Apple Silicon runs Apple Virtualization Framework and does not support Apple ID, I cannot run Software Update from Preferences or use zsh shell to software update the VM. And then it does not support Recovery mode nor USB boot so I can't put the macOS 12.5.1 .ipsw file on USB to upgrade as well.

Does anyone know of a way to upgrade the macOS inside the VM directly, instead of Time Machine the whole disk to a network drive, reinstall with a newer macOS version, and restore the whole disk from network TM share?

p.s. It has been over 1 year and Apple is still seriously lagging proper support on Apple Silicon VM support... sigh.
I am glad that I still have my slow but working Intel MacBook Pro 2019 on hand, even though the M1 MacBook Pro is much faster but seriously crippled when doing multiple configurations development.

**Comment by saagarjha** [comment]
You should be able run through the software update process as usual on virtualized Macs, no Apple ID necessary. Does that not work for you?

**Comment by hanselsin** [comment]
Hi Saagar,

No it doesn’t work. The Software Upgrade keeps complaining missing Internet connection, but indeed everything else is working normally.

FYI this is a guest VM macOS 12.5 created just a few weeks ago on a M1Pro MacBook Pro 16”.

Regards,
Hansel

On 21 Aug 2022, at 11:57 PM, Saagar Jha ***@***.***> wrote:

﻿

You should be able run through the software update process as usual on virtualized Macs, no Apple ID necessary. Does that not work for you?

**Comment by hanselsin** [comment]
Update to the issue. I rolled back the VM image to a week ago (still the same macOS 12.5) and it can software update itself to 12.5.1… dunno what sort of things I screwed up the guest macOS in the past week… anyway the security problem is solved for now.

Regards,
Hansel

On 22 Aug 2022, at 9:32 AM, Hansel SIN ***@***.***> wrote:

﻿ Hi Saagar,

No it doesn’t work. The Software Upgrade keeps complaining missing Internet connection, but indeed everything else is working normally.

FYI this is a guest VM macOS 12.5 created just a few weeks ago on a M1Pro MacBook Pro 16”.

Regards,
Hansel

On 21 Aug 2022, at 11:57 PM, Saagar Jha ***@***.***> wrote:

﻿

You should be able run through the software update process as usual on virtualized Macs, no Apple ID necessary. Does that not work for you?
